### ISSUE TITLE
There is a lot of noise on the surface of the mesh

### ISSUE THREAD

**Issue by ch1998** [Anttwo/SuGaR#39]
I used the lego scene reproduction experiment in nerf_synthetic and successfully extracted the mesh, but the quality seemed unable to reach the quality in the project page.

cmd ： python train.py -s data/nerf_synthetic/lego/ -c output/lego/ -r sdf
Initialize with random point cloud
![image](https://github.com/Anttwo/SuGaR/assets/37291486/5d43db77-546e-4e86-91bc-a79236d6615c)

**Comment by yuedajiong** [comment]
How about -r density?

**Comment by ch1998** [comment]
> How about -r density?

In fact, I also used other data taken by myself for testing, and basically all had noise problems. I will try to use density mode to test again in the future.

**Comment by yuedajiong** [comment]
My understanding, myabe stupid:
   1. Use GS-driection to reconstuct, the discrete (small, big) partiwcles problem is one of the big challanges.
       Compare with direct-Mesh or similar ways, we can think that there is a very strong priori constraint: connected, whole.
   2. Maybe we can use smoothing-loss or other loss to reduce those sand-like things on the surface, iff we know there are not.

**Comment by ch1998** [comment]
Noise points should be generated during the refine stage
The results of the coarse stage do not have obvious noise problems.
use density mode:
result of coarse:
![image](https://github.com/Anttwo/SuGaR/assets/37291486/62c7c678-d5bc-4f94-9974-e1955d125286)

result of refine:
![image](https://github.com/Anttwo/SuGaR/assets/37291486/3778945c-c614-44d0-86d3-1f93095e8a8a)

**Comment by Anttwo** [comment]
Hello ch1998, 

SuGaR was tested on real scenes only, and not on synthetic datasets with masked background (and random initial point cloud). As such, some hyperparameters may not be optimal for synthetic scenes, as they have no background and the spatial extension is extremely limited (and basically just consists in the main object). The lego that you see on the website is the lego from the real "kitchen" scene in the Mip-NeRF360 dataset. 

I have several ideas that could explain your problem:

1. **Is it really a problem with the geometry of the mesh, or with the teture of the mesh?** Looking at the result, it may be a problem of texture. Indeed, the color of the coarse mesh is good-looking, which shouldn't be the case because these are just vertex colors, which usually are not dense enough for producing a good-looking texture. In your case, it looks quite good: This means that you have a really high number of triangles in your scene, probably much more than needed. The UV unwrapping method we use for computing the refined texture is quite dump and for an object with far too many triangles, I think it may produce some small artifacts like the noise you have. Could you try to plot the untextured geometry of your mesh, just to check?
2. **It could be a problem with the geometry of the mesh; In this case, try to reduce the resolution.** For the same reason, this may also be a problem with the geometry. Indeed, if you have far too many triangles, the refinement stage (which slightly moves the vertices to smooth the surface) could produce self-collisions between your triangles, and maybe that could produce the color noise that you see. In the default setting, the scene has 1,000,000 vertices. You can try to use the flag `--low_poly` with `train.py`, this will use the default parameters for a 200,000 vertices mesh. You can also try to manually change the number of vertices with the option `--n_vertices_in_mesh`. Check the README for details about all command line options.

To summarize, you should:
1. Check if it is a problem of texture rather than geometry.
2. If it is a problem of geometry, reducing the resolution of the mesh (either by using `--low_poly` or by adjusting `--n_vertices_in_mesh`)

**Comment by ch1998** [comment]
Thank you very much for your reply. It is precise and clear. I will give it a try.

**Comment by gaoxiangjun** [comment]
> I used the lego scene reproduction experiment in nerf_synthetic and successfully extracted the mesh, but the quality seemed unable to reach the quality in the project page.
> 
> cmd ： python train.py -s data/nerf_synthetic/lego/ -c output/lego/ -r sdf Initialize with random point cloud ![image](https://private-user-images.githubusercontent.com/37291486/292066049-5d43db77-546e-4e86-91bc-a79236d6615c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM0ODg0MDcsIm5iZiI6MTcwMzQ4ODEwNywicGF0aCI6Ii8zNzI5MTQ4Ni8yOTIwNjYwNDktNWQ0M2RiNzctNTQ2ZS00ZTg2LTkxYmMtYTc5MjM2ZDY2MTVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI1VDA3MDgyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjYWQzYmU0MTM3NmYxNjc5ZWIzNmNkNTQ3NDU2Mzg2OGVkNTgxZWE2YThiZTQwOGMyNTZiMmU2ZmIxOTdjZGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g__nzmt5OAbkDwHJdaeUvgMiGjuBdh1cqi-y1Gl_O3s)

I run the `python train.py -s data/nerf_synthetic/lego/ -c outputs/lego/ -r sdf` command but got `FileNotFoundError: [Errno 2] No such file or directory: 'data/nerf_synthetic/lego/images/r_100.png'`，did you meet the similar error and know how to fix it ?

**Comment by ch1998** [comment]
In the nerf_synthetic/lego/ path, copy the test folder, rename it to images, and then run it again

**Comment by gaoxiangjun** [comment]
Thanks for your kind reply, I run it successfully! Many thanks!

**Comment by gaoxiangjun** [comment]
> In the nerf_synthetic/lego/ path, copy the test folder, rename it to images, and then run it again

Did you run successfully of LEGO refine stage? @ch1998 
I got file not found error like `FileNotFoundError: [Errno 2] No such file or directory: 'data/nerf_synthetic/lego/sparse/0/points3D.ply'`

**Comment by clement-chupin** [comment]
@gaoxiangjun Same problem here, any solutions ?

**Comment by danperazzo** [comment]
I got a solution by going to `refine.py` and changing the `line 290`  from:

`ply_path = os.path.join(source_path, "sparse/0/points3D.ply")`

to:

`ply_path = os.path.join(source_path, "points3d.ply")`

Which is the file the 3DGS creates when making the random points for the Blender scenes
